server/routes/mentions.js

```diff
@@ -0,0 +1,76 @@
+const router = require("express").Router();
+
+const requiresAuth = require("./middleware/requiresAuth");
+const { Mention, User } = require("../models");
+
+const callScraper = require("../scraper");
+
+// TODO test mentions in action
+
+router.get("/", requiresAuth, async (req, res) => {
+  const user = await User.findByPk(req.user.id);
+  // note that the getter method for n-m can accept options
+  // as if it were a regular query
+  const mentions = await user.getMentions();
+
+  res.json(mentions);
+});
+
+router.get("/:id", requiresAuth, async (req, res) => {
+  const mention = await Mention.findByPk(req.params.id);
+
+  res.json(mention);
+});
+
+router.post("/", requiresAuth, async (req, res) => {
+  // NOTE I think this is better served as just a test route in order to test the scraper
+  // This shouldn't be placed in the final PR as there will be another means to post new data
+  // into the database.
+  const user = await User.findByPk(req.user.id);
+
+  let mentions = callScraper(req.user.company, 100);
+
+  let result = [];
+  for (let m of mentions) {
+    let userMentions = await user.getMentions({
+      where: {
+        title: m.title,
+      },
```

Comment:
There might be a efficiency concer as it obtains all mentions of this user, but we might want to stop retrieving after we have gotten the first mention, because we only need one mention to know this mention is not a new one.

Reply:
I see what you're saying; this wouldn't be a good solution for very large databases. I can replace this with a more explicit `findOne` call to query based on if there is an existing UserMention entry for that user and a mention with the uniquely identifying attribute. 

This looks to be on unused code that I should've removed before opening the PR, so my mistake. I will remove this route, but the updated routine for the scraper is moved to server/async-worker

Reply:
On the other hand, however, I should point out that user model method `getMentions` only seeks to obtain mentions that are linked to a specific user on the UserMentions through table; it will not seek out on the database in its entirety afaik. I would think it does a chain filter (filters for mentions related to a user, then filter those mentions again that satisfy the query) So in a scenario where we are routinely removing stale mentions, I believe the overall total mentions on a per user basis shouldn't be very high